src/state/service-reader.ts

```diff
@@ -64,7 +64,14 @@ const findLabel = (labels: string[] | undefined, prefix: string) => {
         return undefined;
     }
 
-    const value = labels.find(label => label.startsWith(prefix));
+    const completeLabelKey = !prefix.endsWith(".");
+
+    // Regex which tests the label begins with prefix, if does not end in trailing '.'
+    // assumes the label name is complete therefore expects the label name to be followed
+    // by '='
+    const labelKeyRegex = new RegExp(`^${prefix}${completeLabelKey ? "=" : ""}.+$`);
```

Comment:
Could this be shortened like:
```
if (!prefix.endsWith(".")) { prefix += "="; }
const labelKeyRegex = new RegExp(`^${prefix}.+$`);
```
As a side note, we should probably leave a comment that we are aware of not escaping special chars. I mean, all dots (.) in "prefix" (like any other special chars) will not be interpreted literally in the regex.

Reply:
True - hadn't considered that. I'll add some docs.